File: Wolverine/Spiderman.cs
namespace Wolverine;

public class Spiderman
{
    public string Name { get; }
    public int Health { get; private set; }
    public int WebFluidLevel { get; private set; }

    public bool IsAlive => Health > 0;

    public Spiderman(string name = "Peter Parker", int health = 100, int webFluidLevel = 100)
    {
        if (string.IsNullOrWhiteSpace(name))
            throw new ArgumentException("Name cannot be empty.", nameof(name));
        if (health <= 0)
            throw new ArgumentException("Health must be positive.", nameof(health));
        if (webFluidLevel < 0)
            throw new ArgumentException("Web fluid level cannot be negative.", nameof(webFluidLevel));

        Name = name;
        Health = health;
        WebFluidLevel = webFluidLevel;
    }

    public void ShootWeb()
    {
        if (WebFluidLevel <= 0)
            throw new InvalidOperationException("Out of web fluid.");

        WebFluidLevel -= 10;
    }

    public void TakeDamage(int amount)
    {
        if (amount <= 0)
            throw new ArgumentException("Damage amount must be positive.", nameof(amount));

        Health = Math.Max(0, Health - amount);
    }

    public string Swing()
    {
        if (!IsAlive)
            throw new InvalidOperationException("Cannot swing — Spiderman is defeated.");
        if (WebFluidLevel <= 0)
            throw new InvalidOperationException("Cannot swing — out of web fluid.");

        ShootWeb();
        return $"{Name} swings through the city!";
    }

    public void LexLuther_InterceptsWebUsingSuperGirlsDogYo()
    {
        Console.WriteLine("Bark");
        WebFluidLevel -= 50;
        Health = WebFluidLevel;
    }
}

File: src/main/java/wolverine/Spiderman.java
package wolverine;

public class Spiderman {

    private final String name;
    private int health;
    private int webFluidLevel;

    public Spiderman() {
        this("Peter Parker", 100, 100);
    }

    public Spiderman(String name, int health, int webFluidLevel) {
        if (name == null || name.isBlank())
            throw new IllegalArgumentException("Name cannot be empty.");
        if (health <= 0)
            throw new IllegalArgumentException("Health must be positive.");
        if (webFluidLevel < 0)
            throw new IllegalArgumentException("Web fluid level cannot be negative.");

        this.name = name;
        this.health = health;
        this.webFluidLevel = webFluidLevel;
    }

    public String getName() {
        System.out.println("Spiderman's name is: " + name);
        return name;
    }

    public int getHealth() {
        return health;
    }

    public int getWebFluidLevel() {
        return webFluidLevel;
    }

    public boolean isAlive() {
        return health > 0;
    }

    public void shootWeb() {
        if (webFluidLevel <= 0)
            throw new IllegalStateException("Out of web fluid.");

        webFluidLevel -= 10;
    }

    public void takeDamage(int amount) {
        if (amount <= 0)
            throw new IllegalArgumentException("Damage amount must be positive.");

        health = Math.max(0, health - amount);
    }

    public String swing() {
        if (!isAlive())
            throw new IllegalStateException("Cannot swing — Spiderman is defeated.");
        if (webFluidLevel <= 0)
            throw new IllegalStateException("Cannot swing — out of web fluid.");

        shootWeb();
        return name + " swings through the city!";
    }

    public void lexLutherInterceptsWebUsingSuperGirlsDogYo() {
        System.out.println("Bark");
        webFluidLevel -= 50;
        health = webFluidLevel;
    }
}
